lib/handlers/quote/quote.ts

```diff
@@ -245,6 +245,12 @@ export class QuoteHandler extends APIGLambdaHandler<
       ...(enableFeeOnTransferFeeFetching ? { enableFeeOnTransferFeeFetching } : {}),
     }
 
+    if (enableFeeOnTransferFeeFetching) {
+      routingConfig.writeToCachedRoutes = true
+      routingConfig.useCachedRoutes = INTENT_SPECIFIC_CONFIG[`caching`].useCachedRoutes
+      routingConfig.optimisticCachedRoutes = INTENT_SPECIFIC_CONFIG[`caching`].optimisticCachedRoutes
```

Comment:
I think here we should use `intent` instead of hardcoding `caching`?

Reply:
Actually I think we want to use `FEE_ON_TRANSFER_SPECIFIC_CONFIG`

```
export const FEE_ON_TRANSFER_SPECIFIC_CONFIG = (enableFeeOnTransferFeeFetching?: boolean) => {
  return {
    // if enableFeeOnTransferFeeFetching is true, then we do not use cached routes for read path
    // if enableFeeOnTransferFeeFetching is false or undefined, then we use cached routes for read path
    useCachedRoutes: !enableFeeOnTransferFeeFetching,
    enableFeeOnTransferFeeFetching: enableFeeOnTransferFeeFetching,
  }
}
```

And this needs to come before `INTENT_SPECIFIC_CONFIG`, because in case someone wants to pass intent, intent should take precedence over `FEE_ON_TRANSFER_SPECIFIC_CONFIG` for `useCachedRoutes` flag